### PR TITLE
adds docs for abortOnUnmount

### PR DESCRIPTION
**Pull request by bautistaaa**
Closes [#2632](https://github.com/trpc/trpc/issues/2632)

## 🎯 Changes

Adds docs for `abortOnUnmount`

### Notes
- need help with tests
- removes type annotations that were `any` and will raise a separate issue (https://github.com/trpc/trpc/issues/2648)

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/docs/client/aborting-procedures.md [hunk 1]
```diff
@@ -5,13 +5,53 @@ sidebar_label: Aborting Procedures
...
 
+export const trpc = createTRPCNext<AppRouter, SSRContext>({
+  config() {
+    return {
+      // ...
+      abortOnUnMount: true 
+      // ...
+    };
+  },
+});
```
[the remark this answers is not in the document]
Comment (bautistaaa) [reply www/docs/client/aborting-procedures.md]: `And then add to the createTRPCNext's config-callback docs:`

not sure what you mean by this line. are you saying link to the trpc next docs? or should I just add a @trpc/next section

[the remark this answers is not in the document]
Comment (juliusmarminge) [reply www/docs/client/aborting-procedures.md]: Nah just add the option in that docs as an option? Right now the other options are there? 

https://github.com/trpc/trpc/blob/1ed4127160cf0fcc873494f407de152203bcae01/www/docs/nextjs/introduction.md?plain=1#L213-L221

[the remark this answers is not in the document]
Comment (bautistaaa) [reply www/docs/client/aborting-procedures.md]: ohhh yes, got it thanks

[the remark this answers is not in the document]
Comment (juliusmarminge) [reply www/docs/client/aborting-procedures.md]: But we could ofc also add a note in here like:

`This option can also be passed in the config-callback if you are using @trpc/next, see https://trpc.io/docs/v10/nextjs#config-callback`

**Review by KATT — APPROVED** [review]

**Pull request merged**